File: rest-api/src/main/java/com/fuelstation/managmentapi/fuelorder/domain/FuelOrderService.java
package com.fuelstation.managmentapi.fuelorder.domain;

import com.fuelstation.managmentapi.common.domain.FuelGrade;

public interface FuelOrderService {
    public FuelOrder createFuelOrder(long gasStationId, FuelGrade fuelGrade, float amount);
    public FuelOrder confirmFuelOrder(long fuelOrderId);
    public FuelOrder rejectFuelOrder(long fuelOrderId);
}

File: webgui/src/app/shared/fuel-order/fuel-order.service.ts
import { Observable } from "rxjs"
import { FuelOrder } from "./fuel-order.model"
import { inject, Injectable } from "@angular/core"
import { ApiService, FuelGrade } from "../api"

@Injectable({ providedIn: "root"})
export class FuelOrderService {
      
    private apiService: ApiService = inject(ApiService);
    
    getFuelOrders(): Observable<FuelOrder[]> {
        return this.apiService.get("api/fuel-orders/");
    } 

    getFuelOrderById(fuelOrderId: number): Observable<FuelOrder> {
        return this.apiService.get("api/fuel-orders/" + fuelOrderId);
    }
    
    confirmFuelOrder(fuelOrderId: number) {
        return this.apiService.put("api/fuel-orders/" + fuelOrderId + "/confirm");
    }

    rejectFuelOrder(fuelOrderId: number) {
        return this.apiService.put("api/fuel-orders/" + fuelOrderId + "/reject");
    }

    createFuelOrder(fuelStationId: number, fuelGrade: FuelGrade, amount: number) {
        return this.apiService.post("api/fuel-orders/", { fuelStationId, fuelGrade, amount});
    }

}